src/main/java/com/iota/iri/utils/TransactionTruncator.java

```diff
@@ -0,0 +1,74 @@
+package com.iota.iri.utils;
+
+import com.iota.iri.model.persistables.Transaction;
+
+/**
+ * Provides utility methods to truncate and expand raw transaction data.
+ */
+public class TransactionTruncator {
+
+    /**
+     * The max amount of bytes a signature message fragment is made up from.
+     */
+    public final static int SIG_DATA_MAX_BYTES_LENGTH = 1312;
+    /**
+     * The amount of bytes making up the non signature message fragment part of a transaction.
+     */
+    public final static int NON_SIG_TX_PART_BYTES_LENGTH = 292;
+
+    /**
+     * Truncates the given byte encoded transaction by removing unneeded bytes from the signature message fragment.
+     *
+     * @param txBytes the transaction bytes to truncate
+     * @return an array containing the truncated transaction data
+     */
+    public static byte[] truncateTransaction(byte[] txBytes) {
+        // check how many bytes from the signature can be truncated
+        int bytesToTruncate = 0;
+        for (int i = SIG_DATA_MAX_BYTES_LENGTH - 1; i >= 0; i--) {
+            if (txBytes[i] != 0) {
+                break;
+            }
+            bytesToTruncate++;
+        }
+        // allocate space for truncated tx
+        byte[] truncatedTx = new byte[SIG_DATA_MAX_BYTES_LENGTH - bytesToTruncate + NON_SIG_TX_PART_BYTES_LENGTH];
+        System.arraycopy(txBytes, 0, truncatedTx, 0, SIG_DATA_MAX_BYTES_LENGTH - bytesToTruncate);
+        System.arraycopy(txBytes, SIG_DATA_MAX_BYTES_LENGTH, truncatedTx, SIG_DATA_MAX_BYTES_LENGTH - bytesToTruncate,
+                NON_SIG_TX_PART_BYTES_LENGTH);
+        return truncatedTx;
+    }
+
+    /**
+     * Expands an array containing a truncated transaction using a given reference size
+     * to determine the amount of bytes to pad.
+     *
+     * @param data          the truncated transaction data to be expanded
+     * @param referenceSize the max size to use as a reference to compute the bytes to be added
+     * @return an array containing the expanded transaction data
+     */
+    public static byte[] expandTransaction(byte[] data, int referenceSize) {
+        byte[] txDataBytes = new byte[Transaction.SIZE];
+        int numOfBytesOfSigMsgFragToExpand = referenceSize - data.length;
+        byte[] sigMsgFragPadding = new byte[numOfBytesOfSigMsgFragToExpand];
+        int sigMsgFragBytesToCopy = data.length - TransactionTruncator.NON_SIG_TX_PART_BYTES_LENGTH;

```

Comment:
@luca-moser Tests fail due to this line right here. Now that you have the `expandTransaction()` call taking in variable sizes of transaction you need to also have a conditional here that varies the size of the `sigMsgFragBytesToCopy`. When gossiping you still need the `Protocol.GOSSIP_REQUESTED_TX_HASH_BYTES_LENGTH` to be subtracted from the `data.length`. When expanding for other parts of the code (i.e. `Transaction.read()`) this current logic works.

I tested the following locally and it started gossiping and registering gossiped transactions properly: 
```
if(referenceSize == Transaction.SIZE) {
    sigMsgFragBytesToCopy = data.length - TransactionTruncator.NON_SIG_TX_PART_BYTES_LENGTH;
} else {
    sigMsgFragBytesToCopy = data.length - Protocol.GOSSIP_REQUESTED_TX_HASH_BYTES_LENGTH - TransactionTruncator.NON_SIG_TX_PART_BYTES_LENGTH;
}
```